Documentation/design-docs/framework-version-resolution.md

```diff
@@ -0,0 +1,290 @@
+* `Major` -- Roll forward to lowest higher major version, and lowest minor version, if requested major version is missing. If the requested major version is present, then the `Minor` policy is used.
+* `LatestMinor` -- Roll forward to highest minor version, even if requested minor version is present.
+* `LatestMajor` -- Roll forward to highest major and highest minor version, even if requested major is present.
+* `Disable` -- Do not roll forward. Only bind to specified version. This policy is not recommended for general use since it disables the ability to roll forward to the latest patches. It is only recommended for testing.
+
+Roll-forward setting can be specified in these places:
+1. `.runtimeconfig.json` - `runtimeOptions` property `rollForward` - at this level the setting applies to all framework references in that `.runtimeconfig.json`. For example:
+``` json
+{
+  "runtimeOptions": {
+    "tfm": "netcoreapp3.0",
+    "rollForward": "major",
+    "frameworks": [
+      {
+        "name": "Microsoft.NETCore.App",
+        "version": "3.0.1"
+      }
+    ]
+  }
+}
+```
+
+2. `.runtimeconfig.json` - `framework` property `rollForward` - each framework reference can specify its own setting. For example:
+``` json
+{
+  "runtimeOptions": {
+    "tfm": "netcoreapp3.0",
+    "frameworks": [
+      {
+        "name": "Microsoft.NETCore.App",
+        "version": "3.0.1",
+        "rollForward": "major"
+      }
+    ]
+  }
+}
+```
+
+3. Environment variable `DOTNET_ROLL_FORWARD` - For example:
+```
+set DOTNET_ROLL_FORWARD=LatestMajor
+```
+
+4. Command line argument `--roll-forward` - For example:
+```
+dotnet --roll-forward LatestMinor app.dll
+```
+
+For the framework references specified by the application itself:  
+The order above defines precedence. Later scopes take precedence over earlier ones. So command line wins over everything, environment variable wins over `.runtimeconfig.json` and per-framework setting wins over config-wide setting.
+
+For dependent frameworks:
+Each dependent framework (for example FX1) is loaded with an effective roll-forward value (the one used to resolve that framework). If FX1 has additional framework references those will be loaded using this same roll-forward value as the default, that is unless FX1 specifies different value in its `.runtimeconfig.json` or there's a higher precedence override (environment variable `DOTNET_ROLL_FORWARD` or command line argument `--roll-forward`).
+
+
+## Pre-release versions
+Pre-release version is a version which has a pre-release part, for example `3.0.0-preview4-27415-15`. Everything after the `-` (dash) character is a pre-release identifier. Per [semantic versioning rules](https://semver.org/) pre-release versions are ordered before the same version without any pre-release part. So `3.0.0-preview4-27415-15` comes before `3.0.0`.
+
+### Old behavior
```

Comment:
Instead of this section named "Old behavior", I would call it "Behavior version < 3.0" or "Behavior pre-3.0" and the below as "Behavior version >= 3.0" or "Behavior 3.0 and forward".